### PR TITLE
[stable-2.9] Connect before reading the prompt? (#61797)

### PR DESCRIPTION
**Pull request by Qalthos**
(cherry picked from commit a365e77)

Co-authored-by: Nathaniel Case <ncase@redhat.com>

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

network_cli

Depends-On: https://github.com/ansible/ansible/pull/61899

### CODE REVIEW

**Comment by pabelanger** [comment]
added dependency on iosxr fixes

**Comment by pabelanger** [comment]
recheck

**Comment by abadger** [comment]
A changelog is needed for this and then it can be merged.

**Comment by abadger** [comment]
The problem is that the toplevel of the changelog fragment is a dict, not a list.

File: changelogs/fragments/61797-netcli-get_prompt.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+- bugfixes:
```
Comment (abadger) [inline changelogs/fragments/61797-netcli-get_prompt.yaml]:
```suggestion
bugfixes:
```

**Comment by pabelanger** [comment]
recheck

**Comment by abadger** [comment]
Merged for the 2.9.0rc1 release

**Pull request merged**